bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/WriteFlag.java

```diff
@@ -0,0 +1,70 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.client.api;
+
+import java.util.EnumSet;
+import lombok.Getter;
+
+/**
+ * Flags to specify the behaviour of writes
+ */
+@Getter
+public enum WriteFlag {
+
+    /**
+     * Writes will be acknowledged after writing to the filesystem
+     * but not yet been persisted to disks.
+     */
+    DEFERRED_SYNC(0x1 << 0);
```

Comment:
Is deferred sync implemented yet? We shouldn't expose it to the client until the server side implementation is done.

Reply:
This is a very long story...I have already sent patches for this change but I have to rewrite all.
But I am not moving forward and rewrite the whole stuff without merging one step at a time.
If you want me to send next patches I will do but please first make this patch fully approved by every stakeholder, even without merging

Reply:
This is a very long story...I have already sent patches for this change but I have to rewrite all.
But I am not moving forward and rewrite the whole stuff without merging one step at a time.
If you want me to send next patches I will do but please first make this patch fully approved by every stakeholder, even without merging

Reply:
This is a very long story...I have already sent patches for this change but I have to rewrite all.
But I am not moving forward and rewrite the whole stuff without merging one step at a time.
If you want me to send next patches I will do but please first make this patch fully approved by every stakeholder, even without merging

Reply:
I think it is okay to have this flag in first, since this is the feature planned for 4.7. we can throw UnsupportedException on client if a client is trying to use it. And remove the constraint when the server implementation is in. This should address Ivan's concern here and let @eolivelli move forward with his changes.

Reply:
User visible changes should be the last thing to go in, but I'm fine with this if it gets this patch moving.